### DISCUSSION TITLE
Add columns to store data about the UDP netflow packet

### DISCUSSION THREAD

**Discussion by netixx** [akvorado/akvorado#1523] [Q&A]
Hello,

Currently, we store information about the Exporter in the database (basically based on source address of the netflow packet received)
but we are not keeping any information about the destination of the packet, namely on which Listener it arrived.

On our side we thought of the following examples: 
- Distinguish inside a given device: a given device can export to multiple collectors, for example it could export a set of interfaces (or linecards) to one collector, and another set to another (maybe those are different clients). Those flows may need the same processing, but they may identify overlapping domains and need to be distinguished in the database.
- Identify Netflow packet ingress pipeline: in our case, we have a more complex netflow ingestion pipeline, because each router sends data to 2 gateways  spread in multiple datacenters, and then we consume netflows from one of the gateway, providing failover in case something fails on either the gateway or the path between the device and the gateway.
We would like to be able to identify from which gateway the netflow was acquired. 
For that we had the idea of using the destination address and port as a identifier, our client could then send the flows from gateway A to port A and flows from gateway B to port B (with only one active at any point in time).
If we keep that information in the database, we can then identify the path a given netflow has taken.

Storing the dest IP and dest Port of the netflow packet could also be an easy way to identify (in addition/replacement to Exporter* metadata) the scope associated with this data, eg. datacenter devices send netflows to port X, P device to port Y and PE to port Z.

Would there be interest out there for storing this additional information (dest IP, dest port of the netflow UDP packet) ?

**Comment by vincentbernat** [comment]
I don't quite understand the first case since you already have the ingress and egress information for that.

For the second case, my understanding is that the lookup process for metadata could also use the listener, but maybe it would be easier to just have several instances of the inlet.

If you plan on implementing anything, you may want to wait as I am splitting the inlet into two parts to reduce the work done by the inlet. The second point would need more work in this case.
